CFC_DataCollector/modeinfer/pipeline.py

```diff
@@ -122,13 +159,15 @@ def generateFeatureMatrixAndResultVectorStep(self):
 
       # This will crash the script because we will try to access a record that
       # doesn't exist.
-
       # So we limit the records to the size of the matrix that we have created
       for (i, section) in enumerate(self.confirmedSections.limit(featureMatrix.shape[0]).batch_size(300)):
-        self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
+        #self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
+        featureMatrix[i] = self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
         resultVector[i] = self.getGroundTruthMode(section)
         if i % 100 == 0:
             logging.debug("Processing record %s " % i)
+      featureMatrix = np.nan_to_num(featureMatrix)
```

Comment:
Was is the nan_to_num call here? I had already checked that fix as part of commit 990ea697b363dabcd4493858b4eb6a76929a79d1

Also, `updateFeatureMatrixRowWithSection` already calls np.nan_to_num() in the return statement, so won't this make the call twice?